visidata/loaders/_pandas.py

```diff
@@ -19,25 +21,43 @@ def __getattr__(self, k):
 
 # source=DataFrame
 class PandasSheet(Sheet):
+    def dtype_to_type(self, dtype):
+        import numpy as np
+        try:
+            if np.issubdtype(dtype, np.integer):
+                return int 
+            if np.issubdtype(dtype, np.floating):
+              return float
+            if np.issubdtype(dtype, np.datetime64):
+                return date
+        except TypeError:
+            # For categoricals and other pandas-defined dtypes
+            pass
+        return anytype
+   
+    def read_tsv(self, path, **kwargs):
+        """Partial function for reading TSV files using pd.read_csv"""
+        import pandas as pd
+        return pd.read_csv(path, sep='\t', **kwargs)
+
     def reload(self):
-        import numpy
-        import pandas
-        def dtypeToType(df, colname):
-            t = df[colname].dtype
-            if t == numpy.int64: return int
-            if t == numpy.float64: return float
-#            if t == pandas.Timestamp: return date
-            return anytype
-
-        if isinstance(self.source, pandas.DataFrame):
+        import pandas as pd
+        if isinstance(self.source, pd.DataFrame):
             self.df = self.source
         elif isinstance(self.source, Path):
             filetype = getattr(self, 'filetype', self.source.ext[1:])
-            readfunc = getattr(pandas, 'read_'+filetype) or error('no pandas.read_'+filetype)
+            if filetype == 'tsv':
+                readfunc = self.read_tsv 
+            else:
+                readfunc = getattr(pd, 'read_'+filetype) or error('no pandas.read_'+filetype)
             self.df = readfunc(self.source.resolve(), **options('pandas_'+filetype+'_'))
 
-        self.columns = [ColumnItem(col, type=dtypeToType(self.df, col)) for col in self.df.columns]
+        # TODO: should we reset the index here and add it as a key column?
```

Comment:
Ah, this is exactly what @danlat was suggesting in his PR.  I'm not so familiar with pandas DF and its index, but it sounds like it's something that pandas users want/need access to.  So let's do that (maybe merging them both in separately).

Reply:
It's similar to a database index column in that it's meant to be sorted and unique for quick access and querying, but pandas does not include the index in DataFrame.columns (from the list comprehension in this code)